### ISSUE TITLE
gitattributes gets ignored on checkout

### ISSUE THREAD

**Issue by GabriRuflex** [git-for-windows/git#3178]
- [X] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.31.1.windows.1
cpu: x86_64
built from commit: c5f0be26a7e3846e3b6268d1c6c4800d838c6bbb
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Versione 10.0.17763.1817]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: Notepad++
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Core
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

None

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git clone remoteRepoUrl
cd repoName
git checkout  -b branchX origin/branchX
```
 - What did you expect to occur after running these commands?

.gitattributes must not be ignored, its content is:
```
* -text
```

 - What actually happened instead?

.gitattributes is ignored.

It isn't ignored if:

I put it into the master branch (the first one checked by git clone command)
or
I use the command `git clone --branch branchX remoteRepoUrl`

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Private repository

**Comment by siprbaum** [comment]
The .gitattributes file is always per branch you checkout. 
So if the other branchX you mentioned didn't contain a .gitattribues file, then there is nothing for git to use.

**Comment by GabriRuflex** [comment]
Yes I know, but the branchX actually contains its own gitattributes file, which is ignored.

**Comment by dscho** [comment]
This is seems not to be a Windows-specific issue. Please take it to [the Git mailing list](mailto:git@vger.kernel.org) (send plain-text messages, HTML messages are dropped silently).

**Comment by tboegi** [comment]
It would be helpful to come up with a receipt how to produce it.
A short set of instructions, please, like this:
https://stackoverflow.com/help/minimal-reproducible-example
